### ISSUE TITLE
Slow Inference Speed (GPU Mode)

### ISSUE THREAD

**Issue by riven314** [montrealrobotics/ENetDepth#4]
Hi again, 

I intend to deploy your model in real time, so I tried to time the inference speed of your model (ENetDepth). I fonud that it took 0.05 - 0.06 s per inference with GPU for image of size (240, 320). Though it is acceptable for real time purpose, but it has a gap to the speed u report. (40Hz) Do you have any idea how I can speed up the model inference?

For the sake of speed analysis, I have uploaded the code in another repo for easier replication: [https://github.com/riven314/ENetDepth_Tmp](url). You can simply run webcam_predict.py to get the average speed result. Below is a sample output of the script:
![slow_speed](https://user-images.githubusercontent.com/21143399/66506864-b58db080-eb00-11e9-8d84-bb36feee00f6.JPG)

A bit details about my hardware spec: I am using Lenovo Y740 notebook with GeForce RTX 2070 Max Q.

**Comment by riven314** [comment]
(apology for the exaggerated issue topic, the model has moderate inference speed in GPU, but it should hv a faster speed i believe)

**Comment by krrish94** [comment]
Just took a quick look at your repo. I'd say the times you report are 2x slower than what they were on my end. Although I'm not super sure why this'd be the case, here's what I can think of:

1. Latency in calling `model.eval()` each time?
2. I'd also assume the pytorch dataloaders are much faster at loading data (as they multiprocess).

It'll be worth timing regular pytorch dataloaders. (If you're doing webcam stuff, probably use the new torchvision loaders that directly transfer images to gpu)

**Comment by riven314** [comment]
I tried to time the inference speed again, this time:
1. I added 3 test cases (small image case, medium image case and large image case)
2. time the GPU inference computation with synchronization
3. only pass model.eval() once

This time, the speed seems faster than what I recorded last time:
![overall_speed](https://user-images.githubusercontent.com/21143399/66715038-28dd3e00-edf1-11e9-8279-a0377522798f.JPG)

I tried to time it again without (2), (3), the yielded time is similar. So I still can't figure out why I got relatively slow speed last time. (anyway)
And surprisingly, I found that the model could give reasonable inference speed even for large image (e.g. 720, 1280, 3 as example).
Now, I am finding place for further speed optimization / improvement on stability of speed

Remark:
- updated repo for your reference: [https://github.com/riven314/ENetDepth_TimeAnlysis_Tmp](url)
the main script is still webcam_predict.py
- the recorded screenshot has uneven number of trials for 3 test cases. It is probably due to the synchronization operation (I guess)
